Source/WebCore/svg/SVGPathElement.cpp

```diff
@@ -223,22 +223,22 @@ RenderPtr<RenderElement> SVGPathElement::createElementRenderer(RenderStyle&& sty
 
 Path SVGPathElement::path() const
 {
-    if (auto* renderer = this->renderer()) {
-        if (auto* basicShapePath = renderer->style().d())
+    if (CheckedPtr renderer = this->renderer()) {
+        if (RefPtr basicShapePath = renderer->style().d())
             return basicShapePath->path({ });
     }
 
-    return m_pathSegList->currentPath();
+    return Ref { m_pathSegList }->currentPath();
 }
 
 void SVGPathElement::collectPresentationalHintsForAttribute(const QualifiedName& name, const AtomString& value, MutableStyleProperties& style)
 {
     if (name == SVGNames::dAttr && document().settings().cssDPropertyEnabled()) {
         // In the case of the `d` property, we want to avoid providing a string value since it will require
         // the path data to be parsed again and path data can be unwieldy.
-        auto property = cssPropertyIdForSVGAttributeName(name, document().settings());
+        auto property = cssPropertyIdForSVGAttributeName(name, Ref { document().settings() });
```

Comment:
The style I usually see is to call `protectedSettings()` on Document here, but `Ref { x }` shows up often too, so maybe that’s also fine. @rniwa or @cdumez might have an opinion.

Reply:
Did not know about `protectedSettings()`! I'll adopt whatever the braintrust suggests.

Reply:
Yeah, the preferred style here is to call `protectedSettings()`. An explicit instantiation of `Ref` or `RefPtr` is useful when we're allocating a local variable. In the cases where we use the result of a function call to call another function or chain a function call (e.g. `document().settings().nonTrivialFunction()`), a preferred method is to use `protectedX` function.